index.js

```diff
@@ -1,18 +1,131 @@
+  console.log('FYI, you entered: ' + secretNumber + ' (Again, this will be kept secret until the end!)'); 
+
+if ((secretNumber >= min) && (secretNumber <= max)) {   
+
+// mainGuessing 
+  cpuGuessNum = getRandomIntInclusive(min, max);   // Guessing a random number in the range to start
+  console.log('(This program currently randomly guesses only the first try) Is it... ' + cpuGuessNum); 
+  let pgmTries = 1; 
+  let userRespYorN = await ask('? (Please type a Y or a N only) -> ');
+    let userRespYorNUC = userRespYorN.toUpperCase().trim();  // to capitalize letters and also trim spaces
+    
+  if (userRespYorNUC === 'Y') {
+    console.log('You stated that this was the correct number, Yea!');  
+    console.log('The program has now tried: ' + pgmTries + ' times.'); // game exits since it's done
+    process.exit();
+  } else if (userRespYorNUC === 'N') {
+    console.log('You stated it was NOT correct, or ' + userRespYorNUC + ' , Sorry.');
+    // keepGuessing
+
+  pgmTries = pgmTries + 1;  // add another try
+  console.log('The program will try another time - This time makes: ' + pgmTries + ' times.');
+  console.log('The maximum number of tries this program should theoretically take is only: ' + maxTries);
+
+  // it was a N, then continue - 
+  if (userRespYorNUC === 'N') { 
+    continueTrying = (true);}    
+  let userRespHorL = await ask("Is it higher (H) or lower (L)? ");
+    
+  // in case the user used lower case - 
+  let userRespHorLUC = userRespHorL.toUpperCase();  
+    
+  // goes to High/Low Logic -  
+    if (userRespHorLUC == 'H') {  // goes to Higher logic 
+   (gotoHighLoop = (true)); 
+     // get a Higher range --
+     let newMin = (cpuGuessNum + 1); // causes it to not include cur low guess
+     let newRange = (max - newMin);
+     cpuGuessNum = (newRange/2);
+  } else if (userRespHorLUC == 'L') {  // goes to Lower Logic
+    (gotoLowLoop = (true));
+    // get a lower range --
+     let newMax = (cpuGuessNum - 1); // causes it to no include cur high guess 
+     let newRange = (newMax - min);
+     cpuGuessNum = (newRange/2);
+     
+    // else needs to go to end of pgm after - 
+  } else {
+   cheatMsgUnexpectedErr(); // putting this in for messages that don't equal y or n
+  }
+   
+} else {
+  cheatMsg1();
+}  
+
+} else { 
+  cheatMsgRangeErr();
+}  
+
+endGameMsg(); // totally ends out of the game
+process.exit();
+}
+
+// -----------putting various message functions and exit functions below this line -----------
+
+function cheatMsg1() {
+  console.log('Sorry, something did not compute!');
+  endGameMsg();
   process.exit();
 }
+
+/* function cheatMsg2() {
+  console.log('Sorry, something did not compute! You entered: ' + userRespHorL + 'or we tried to compute: ' + userReponseHorLUC);
+} */  // commenting out for now until more code is written
+
+function endGameMsg() {
+  console.log("We're done then. Goodbye."); // temporarily ending game after one random try
```

Comment:
You're on the right track, but as of now, this function is hitting no matter what the user inputs following the initial "Is it higher (H) or lower (L)" question. Also, if I were to comment it out, you're going to get stuck in an infinite loop.